Automattic/wp-calypso
# Plans: Fix plans flashing incorrect interval

Closes to #75778

## Proposed Changes
The issue is due to the fact that when clicking `Get Business` the url gets updated with the domain step and the plans component re-renders, The `getIntervalType` function then checks the url for query param `intervalType` which doesn't exist since url is updated to the domain step, it then fallback to yearly because `intervalType` is missing.


* Since the plan step url is stored in the `path` props, this change passes the path to the `getIntervalType` method and extracts the interval from the `path`. If it doesn't exist in the path it attempts to find it in `window.location` otherwise it fallback to yearly.

## Testing Instructions
1. Browse `/start/hosting`;
2. Switch to monthly plans;
3. Click any monthly plan;
4. Check that the screen doesn't flash the annual plans before proceeding to the domains step. (see issue for video of what's happening)



*

## Pre-merge Checklist
